stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -37,7 +43,22 @@ internal class PaymentOptionsViewModel(
         selection.value?.let { paymentSelection ->
             eventReporter.onSelectPaymentOption(paymentSelection)
             prefsRepository.savePaymentSelection(paymentSelection)
-            _userSelection.value = paymentSelection
+            processSelection(paymentSelection)
+        }
+    }
+
+    private fun processSelection(paymentSelection: PaymentSelection) {
+        val requestSaveNewCard = (paymentSelection as? PaymentSelection.New)?.shouldSavePaymentMethod
+            ?: false
+
+        if ((customerConfig != null) && requestSaveNewCard) {
+            _viewState.value = ViewState.PaymentOptions.Completed(
+                PaymentOptionResult.Succeeded(paymentSelection)
+            )
+        } else {
+            _viewState.value = ViewState.PaymentOptions.Finished(
+                PaymentOptionResult.Succeeded(paymentSelection)
+            )
```

Comment:
Can you add a comment here that says "we need to create the payment method and attach to customer"?

Reply:
Also seems like we should just not emit `ViewState.PaymentOptions.Completed` until the card has been attached to the customer if needed, obviating the need for `Finished`?